File: ios/BigIntfile.m
//
//  BigIntfile.m
//  RNdemo
//
//  Created by TAOLIU on 5/5/21.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

//to expose methods to react native
@interface RCT_EXTERN_MODULE(BigIntDemo,NSObject)
//methods in BigIntfile.swift
RCT_EXTERN_METHOD(getKey)
RCT_EXTERN_METHOD(callbackToGetKey: (RCTResponseSenderBlock)callback)
@end

File: ios/BigIntfile.swift
//
//  BigIntfile.swift
//  RNdemo
//
//  Created by TAOLIU on 5/5/21.
//

import Foundation
import BigInt

@objc(BigIntDemo) class BigIntDemo: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool {return true}
  
  @objc static var privateKeyResult = ""
  
  @objc func getKey() {
    
    let result = DHKeyExchangeManager.generateDHExchangeKeys(generator: "9", primeNumber: "1");
    print("iOS end >> privatekey =  " + result.privateKey! )
    
    BigIntDemo.privateKeyResult = result.privateKey!
    
  }

  @objc func callbackToGetKey(_ callback: RCTResponseSenderBlock) {
    callback([NSNull(), BigIntDemo.privateKeyResult])
  }
  
}



